src/transaction/bundlr.rs

```diff
@@ -32,38 +32,46 @@ pub struct BundlrTx {
 }
 
 impl BundlrTx {
-    pub fn new(target: Vec<u8>, data: Vec<u8>, tags: Vec<Tag>) -> Self {
+    pub fn new(target: Vec<u8>, data: Vec<u8>, tags: Vec<Tag>) -> Result<Self, BundlrError> {
         let mut randoms: [u8; 32] = [0; 32];
         let sr = ring::rand::SystemRandom::new();
-        sr.fill(&mut randoms).unwrap();
+        match sr.fill(&mut randoms) {
+            Ok(()) => (),
+            Err(err) => return Err(BundlrError::Unknown(err.to_string())),
+        }
         let anchor = randoms.to_vec();
 
-        BundlrTx {
+        Ok(BundlrTx {
             signature_type: SignerMap::None,
             signature: vec![],
             owner: vec![],
             target,
             anchor,
             tags,
             data: Data::Bytes(data),
-        }
+        })
     }
 
     fn from_info_bytes(buffer: &[u8]) -> Result<(Self, usize), BundlrError> {
         let sig_type_b = &buffer[0..2];
-        let signature_type = u16::from_le_bytes(<[u8; 2]>::try_from(sig_type_b).unwrap());
+        let signature_type = u16::from_le_bytes(
+            <[u8; 2]>::try_from(sig_type_b)
+                .map_err(|err| BundlrError::BytesError(err.to_string()))?,
```

Comment:
I would move this to a separate variable and pass it here. Would make the code a bit cleaner.